### PR TITLE
changelog: update for 1.23.0 release

### PR DESCRIPTION
**Pull request by yeazelm**
**Issue number:**

Related to #4190

**Description of changes:**
Update the changelog for the 1.23.0 release.

**Testing done:**
N/A

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by ginglis13 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+
+### Kubernetes
+* Support Kubernetes NVIDIA Device Plugin configurations through API ([#4182])
+* Support NVIDIA Container Toolkit configurations through API ([#4182])
+
+## Build Changes
+* Update bottlerocket-sdk to 0.45 ([#4189])
+
+### Kit Changes
+* Update to bottlerocket-core-kit to v2.5.0 ([#4189])
```
Comment (larvacea) [inline CHANGELOG.md]: Nit: "Update to ... to" is awkward. Removing the first "to" would make it resemble "Update bottlerocket-sdk to...."

**Review by larvacea — APPROVED** [review]

**Review by arnaldo2792 — APPROVED** [review]

**Comment by arnaldo2792** [comment]
We want to include #4203

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,33 @@
...
+* Support NVIDIA Container Toolkit configurations through API ([#4182])
+
+## Build Changes
+* Update bottlerocket-sdk to 0.45 ([#4189])
+
+### Twoliter
+* Update bottlerocket-core-kit to v2.5.0 ([#4189])
+* Perform binary checksum validation ([#4192])
+* Update Twoliter to v0.4.6 ([#4200])
+* Update bottlerocket-core-kit to v2.6.0 ([#4203])
```
Comment (vyaghras) [inline CHANGELOG.md]: Should we merge both Core kit releases and say "Update bottlerocket-core-kit to v2.6.0 ([#4189], [#4203])"

**Comment by piyush-jena** [comment]
Apart from @vyaghras ' comment, nothing else to call out.

**Review by piyush-jena — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (arnaldo2792) [reply CHANGELOG.md]: I guess I can do that without calling out the version

Comment (piyush-jena) [reply CHANGELOG.md]: While updating the website I check the changelog to get the core-kit version. I would prefer the mention of the version in one of the lines.

Comment (piyush-jena) [reply CHANGELOG.md]: Reference: https://github.com/bottlerocket-os/bottlerocket/pull/4157

**Pull request merged**